grid2op/Reward/AlertReward.py

```diff
@@ -0,0 +1,263 @@
+
+
+    """
+
+    def __init__(self, logger=None, reward_min_no_blackout=-1.0, reward_min_blackout=-10.0, 
+                 reward_max_no_blackout=1.0, reward_max_blackout=2.0, reward_end_episode_bonus=1.0):
+        BaseReward.__init__(self, logger=logger)
+        # required if you want to design a custom reward taking into account the
+        # alert feature
+        self.has_alert_component = True
+        self.is_alert_used = False  # required to update it in __call__ !!
+
+        
+        self.reward_min_no_blackout = dt_float(reward_min_no_blackout)
+        self.reward_min_blackout = dt_float(reward_min_blackout)
+        self.reward_max_no_blackout = dt_float(reward_max_no_blackout)
+        self.reward_max_blackout = dt_float(reward_max_blackout)
+        self.reward_end_episode_bonus = dt_float(reward_end_episode_bonus)
+        self.reward_no_game_over = dt_float(0.0)
+
+        self.total_time_steps = dt_float(0.0)
+        self.time_window = None
+
+        self._has_attack_in_time_window = False # MaJ à chaque appel du call 
+        self._fist_attack_step_in_time_window = None # MaJ à chaque appel du call 
+        self._attacks_in_time_window = None # MaJ à chaque appel du call 
+        self._has_attack_at_first_time_in_window = None # MaJ à chaque appel du call 
+        self._is_first_step_of_attack = None # MaJ à chaque appel du call 
+
+        self._has_line_alerts_in_time_window = False
+        self._line_alerts_in_time_window = None # pas de temps en ligne et lignes élec en colonnes 
+        
+        # index à quel pas de temps on est 
+        self.current_index = dt_int(-1)
+        self.current_step_first_encountered = dt_int(-1) # à ne mettre à jour que lorsque le current_step passe au suivant (la première fois)
+        # Pour ne MaJ qu'un pas de temps 
+
+        self.reward_unit_step = dt_int(-1)
+        
+        self.last_attacked_lines = None
+        self.has_new_attack = False
+
+    def initialize(self, env):
+        from grid2op.Environment._ObsEnv import (
+            _ObsEnv,
+        )  # to avoid circular dependencies
+
+        # This reward is not compatible with simulations
+        if isinstance(env, _ObsEnv):
+            raise Grid2OpException(
+                'Impossible to use the "AlertReward" with a simulation environment ``_ObsEnv``.' 
+                'Please make sure "env._has_attention_budget" is set to ``True`` or '
+                "change the reward class with `grid2op.make(..., reward_class=AnyOtherReward)`"
+            )
+        if not env._has_attention_budget:
+            raise Grid2OpException(
+                'Impossible to use the "AlertReward" with an environment for which this feature '
+                'is disabled. Please make sure "env._has_attention_budget" is set to ``True`` or '
+                "change the reward class with `grid2op.make(..., reward_class=AnyOtherReward)`"
+            )
+        if env.parameters.ASSISTANT_WARNING_TYPE != "BY_LINE":
```

Comment:
Same here, should not be env.parameters, but directly an attribute of the environment.